File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java

class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {

        return numberToCheck == String.valueOf(numberToCheck).chars().map(Character::getNumericValue).mapToDouble(c -> Math.pow(c, String.valueOf(numberToCheck).length())).sum();
    }
}

File: fortran/armstrong-numbers/armstrong_numbers.f90
module armstrong_numbers
  implicit none
contains
  logical function isArmstrongNumber(num)
    integer, intent(in) :: num
    integer :: sum, i, digit, len
    character(10) :: digits
    isArmstrongNumber = .false.
    write(digits,"(i10)") num
    digits = trim(adjustl(digits))
    len = len_trim(digits)
    sum = 0
    do i = 1, len
      read(digits(i:i), *) digit
      sum = sum + digit ** len
    end do
    if (sum == num) then
      isArmstrongNumber = .true.
    end if
    return
  end function
end module

